projects/ngrx-auto-entity/src/lib/util/selector-map-builder.spec.ts

```diff
@@ -14,10 +14,48 @@ import {
   mapToDeletedAt,
   mapToEntityArray,
   mapToLoadedAt,
+  mapToReplacedAt,
   mapToSavedAt,
-  mapToSortedEntityArray
+  mapToSortedEntityArray,
+  mapToUpdatedAt
 } from './selector-map-builder';
 
+expect.extend({
+  toBeMemoizedSelector(received) {
+    const isFunction = typeof received === 'function';
+    const hasProjector = !!received.projector;
+    const hasRelease = !!received.release;
+    const hasSetResult = !!received.setResult;
+
+    const pass = isFunction && hasProjector && hasRelease && hasSetResult;
+
+    const opts = { isNot: this ? this.isNot : false };
+    const message = pass
+      ? () => this.utils.matcherHint('toBeMemoizedSelector', undefined, undefined, opts) +
+        '\n\n' +
+        `Expected: not [Function memoized]` +
+        `Received: ${this.utils.printReceived(received)}`
+      : () => this.utils.matcherHint('toBeMemoizedSelector', undefined, undefined, opts) +
+        '\n\n' +
+        `Expected: [Function memoized]` +
+        `Received: ${this.utils.printReceived(received)}`;
+
+    return { actual: received, message, pass };
+  }
+});
+
+declare global {
+  namespace jest {
+    interface Matchers<R> {
+      toBeMemoizedSelector(): R;
+    }
+
+    interface Expect {
+      toBeMemoizedSelector(): any;
+    }
+  }
+}
```

Comment:
I feel that this shouldn't be duplicated 3 times throughout the project. This could be shared via a `__tests__` folder or something similar say `__matchers__`.

Reply:
I tried to centralize it...and TS complained about the typing declarations. I tried to centralize just the expect.extend, and TS complained about the typing declarations.... Sometimes TS is weird. :P